networkx/algorithms/bipartite/edge_coloring.py

```diff
@@ -0,0 +1,279 @@
+""" Functions related to Edge Coloring"""
+
+import networkx as nx
+
+__all__ = ["edge_coloring"]
+
+
+@nx.utils.not_implemented_for("directed")
+@nx.utils.not_implemented_for("multigraph")
+@nx._dispatchable(name="bipartite_edge_coloring")
+def edge_coloring(G, top_nodes=None, strategy="kempe-chain"):
+    """Returns a minimum edge coloring of the bipartite graph `G`.
```

Comment:
Yes, this is the coloring with minimum possible number of colors, hence naming the function as `minimum_edge_coloring` is better. But for convenience we have also added an alias `edge_coloring` for the same.

Reply:
We changed the function name to `minimum_edge_coloring` as the name describes the function well. But for convenience we have also added an alias called `edge_coloring` for convenience.

Reply:
> hence naming the function as `minimum_edge_coloring` is better. But for convenience we have also added an alias `edge_coloring` for the same.

I'm curious about best practice for adding aliases in NetworkX. I know `maximum_matching` is an alias for `hopcroft_karp_matching`, which is mentioned in their docstring. I don't know of other aliases, but I could probably find them programmatically.

I'm always a little bit surprised whenever I see aliases, and I expect an alias would be added when a function is heavily used (and names obvious) so as to justify the convenience. I'm not familiar with this algorithm, so my uninformed knee-jerk reaction would be to prefer to avoid aliasing this function and to choose one name such as `minimum_edge_coloring`.